ui/app/components/app/modals/qr-scanner/qr-scanner.component.js

```diff
@@ -1,99 +1,139 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import log from 'loglevel'
 import { BrowserQRCodeReader } from '@zxing/library'
+import { getEnvironmentType } from '../../../../../../app/scripts/lib/util'
+import { ENVIRONMENT_TYPE_FULLSCREEN } from '../../../../../../app/scripts/lib/enums'
 import Spinner from '../../../ui/spinner'
 import WebcamUtils from '../../../../../lib/webcam-utils'
 import PageContainerFooter from '../../../ui/page-container/page-container-footer/page-container-footer.component'
 
+const READY_STATE = {
+  ACCESSING_CAMERA: 'ACCESSING_CAMERA',
+  NEED_TO_ALLOW_ACCESS: 'NEED_TO_ALLOW_ACCESS',
+  READY: 'READY',
+}
+
 export default class QrScanner extends Component {
   static propTypes = {
     hideModal: PropTypes.func.isRequired,
     qrCodeDetected: PropTypes.func,
-    scanQrCode: PropTypes.func,
-    error: PropTypes.bool,
-    errorType: PropTypes.string,
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
   constructor (props, context) {
-    super(props)
+    super(props, context)
```

Comment:
I'd rather we drop the context param instead, I don't know that I've seen two args passed here before.

Reply:
Sure, I guess it's not required. Apparently it's needed if you're using `this.context` within the constructor though (e.g. https://github.com/facebook/react/issues/13944#issuecomment-432858365 )

I pass it in out of habit, at least for any constructors that have `contextTypes`.